File: 647-palindromic-substrings/647-palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        count = 0
        for i in range(len(s)):
            start, end = i, i
            while start >= 0 and end < len(s) and s[start] == s[end]:
                count += 1
                start -= 1
                end += 1
            
            start, end = i, i+1
            while start >= 0 and end < len(s) and s[start] == s[end]:
                count += 1
                start -= 1
                end += 1
                
        return count

File: 647-palindromic-substrings/647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String str) {
        char[] s = str.toCharArray();
        int len = s.length;
        int count = 0;
        for(int i = 0; i < len; i++){
            for(int start = i, end = i; start >= 0 && end < len && s[start--] == s[end++]; )   count++;
            for(int start = i, end = i+1; start >= 0 && end < len && s[start--] == s[end++]; )   count++;
        }
        return count;
    }
}